resources/app/log-collector.yaml

```diff
@@ -0,0 +1,146 @@
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: log-collector
+  namespace: kube-system
+  labels:
+    app: log-collector
+spec:
+  type: ClusterIP
+  ports:
+    - name: api
+      protocol: TCP
+      port: 8083
+  selector:
+    app: log-collector
+
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: log-forwarders
+  namespace: kube-system
+
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: log-collector
+  namespace: kube-system
+  labels:
+    app: log-collector
+data:
+  adapter.json: |
+    {
+      "Gravity": {
+        "ApiListenAddr": "0.0.0.0:8083",
+        "Kubernetes": {
+          "Namespace": "kube-system",
+          "ForwarderConfigMapName": "log-forwarders"
+        }
+      },
+
+      "Logrange": {
+        "Partition": "logrange.pipe=__default__",
+        "ForwarderTmplFile": "/opt/logrange/gravity/config/forward-tmpl.json",
+        "CronQueries": [
+          {
+            "Query": "truncate minsize 4mb maxsize 500mb maxdbsize 10gb",
```

Comment:
Could you explain this syntax a bit more please? Eg what's 4MB/500MB/10GB? Isn't 10GB a bit too "generous" for the logs? How much data approximately is that?

Reply:
**minsize 4mb** - every partition (unique combination of tags) won't be truncated to less than 4 mb, e.g. partition in k8s is a combination of tags: ns (namespace), pod (pod name), cname (container name), cid (container id)...

**maxsize 500mb** - every partition won't be more than 500mb, 'truncate' cronQuery is responsible for this trimming...

**maxdbsize 10gb**  - the whole database won't be more than 10gb (this includes LogEvents, i.e. raw message data + attached tags), again 'truncate' cronQuery is responsible for this trimming...

It's more or less up to you what amount of data you want to store, we can make it less/bigger...
In k8s configuration it can be estimated by taking into account that we store the raw message + 4 tags mentioned above...